### ISSUE TITLE
How many hard disk would be taken?

### ISSUE THREAD

**Issue by lamz138138** [marbl/canu#405]
Hi, I had saw a related issue (https://github.com/marbl/canu/issues/193), but I think my question is a little different, so I open a new issue.

I had assembly my genome that about 3G, you can image it as human. I had used 8X CLR reads (mean length is 11K), and 30X CCS reads (I am using their subreads, mean length is 4K), but now it already take 11T hard disk (now it in the step of correction/*.ovlStore.BUILDING), I think this is abnormal, since it said about 3T in  (https://github.com/marbl/canu/issues/193), do you think it is because there are a lot of short CCS subreads? I will have another 30X CLR data (so 38X CLR reads), if I re-run the pipeline, do you think it would take less hard disk? Also, what files and what the order could I delete in the progress so I can save hard disk?

Besides, with above data, it had take two weeks, I feel too slow, what is the recommend machine? My cluster have 20 small node (16 cores, 64G memory), 3 big node (two with 64 cores, one 32 cores, all 512G memory), would it take a long time to assembly with 60X and 100X data?

my command list as follow:
```
$canuPath/canu -p canu -d canu \
        genomeSize=3g \
        gridEngineThreadsOption="-pe myPE THREADS" \
        gridEngineMemoryOption="-l mf=MEMORY" \
        gridOptionsJobName="canuJob" \
        java=/tools/java/bin/java \
        gridOptions="-V -q myQ -S /bin/bash" \
        gnuplotImageFormat=svg \
        ovsMemory=2-300g\
        -pacbio-raw data/CCS.fq \
        -pacbio-raw data/CLR_1.fq \
        -pacbio-raw data/CLR_2.fq \
        >canu.log 2>canu.err
```

Any suggestion would be grateful!
Best wishes!

**Comment by brianwalenz** [comment]
Mixing CLR (uncorrected) and CCS (corrected) isn't supported this way.  Canu will try to 'correct' the CCS reads again.

You can try correcting just the 8x of CLR (see the FAQ for low coverage settings) but I don't think it'll work very well.  Correcting 38x of CLR reads should work well (so should assembly):

`canu -correct ... -pacbio-raw CLR*fq`

Once the CLR are corrected, you can then assemble them with the CCS:

`canu -trim-assemble ... -pacbio-corrected CCS.fq correctedReads.fasta.gz`

Number of overlaps (and thus size) is related to both number of reads and repeat content.  11TB for human might be bit on the large side - this could be caused by the CCS reads as the overlaps generated for correction are tuned for noisy reads.

**Comment by skoren** [comment]
Mostly, what @brianwalenz said. I definitely wouldn't include CCS reads during correction, they're probably not going to be very useful for assembly in general once you have 38X CLR.

 I'll just add that the major determinant of space used is the repetitiveness of the genome. If your genome is the same size as human but more repetitive (like say a plant) it will take more space. You're near the peak space usage (and it could be decreased by erasing some intermediate files) at the store creation. If you look at the size of files in 1-overlapper/results, you can estimate the total space used as approximately double that. During store construction there are two copies of every overlap, once the store is build the redundant copy is cleaned up.

**Comment by lamz138138** [comment]
Hi， thanks for reply!

I am sorry that I didn't make it clean, although I name it CCS.fq, they are just high-error subreads, not high-quality CCS reads.

**Comment by lamz138138** [comment]
Hi, so you all think including the subreads of CCS reads wouldn't be help? I had hoped that exists some locus, where have only one CLR reads, but there are several CCS subreads in there.

Once canu.ovlStore.BUILDING finished, can I delete 1-overlapper/results and how much storage would canu.ovlStore.BUILDING take? I only left 1.3T now, should I stop the program.

Best wishes!

**Comment by skoren** [comment]
First, we should clarify what you mean when you say CCS. CCS reads are generated by the SMRTportal as consensus of multiple passes of a single SMRTbell during sequencing (http://files.pacb.com/software/smrtanalysis/2.2.0/doc/smrtportal/help/!SSL!/Webhelp/Portal_PacBio_Glossary.htm). You said in an earlier message you meant they are just short CLR reads not multi-pass reads? If they are just short reads, but single-pass then you should include them in correction as they would only have 85-90% identity. If they are consensus from multi-pass then they would have >97% identity so correction doesn't make sense. 

You can delete 1-overlapper/blocks now. Canu will delete 1-overlapper/results for you when the store construction is done. How much space is 1-overlapper/results taking now and what step is the overlap store construction on (what is the output of the last canu.\*.out file in canu-scripts). If, based on that, you are going to run out of space, it is possible to erase 1-overlapper/results before the store construction is fully complete but this risks you have to restart from scratch if anything goes wrong during store construction. This is why the default behavior is to only remove the results folder after the store is constructed.

**Comment by lamz138138** [comment]
Hi, thanks for reply!

When I mention CCS subreads, I mean short CLR reads with 85%~90% identity, in contrast, when I say CCS reads, it is 95% identity. I made this type of reads previously for some reason, but I don't know whether should I include high-error CCS subreads during assembly since I only have 38X long CLR reads, whether they would be help? In fact, since there are about 8 pass for each CCS reads, the 30X CCS equal to 4X coverage only, should I drop them out?

Any suggestion would be grateful!

**Comment by lamz138138** [comment]
I had move some of data to another hard disk, and make soft link now. The repeat is same as human, so it consume a lot of hard disk because there are too many overlap between CCS reads? Using 38X CLR reads would be take less storage?

**Comment by brianwalenz** [comment]
I'd just give up on the current assembly.  It sounds like its at the point where it's going to be a bunch of trouble to get finished, and I'm not sure what Canu will do with CCS subreads.  The issue is that there are many reads that are essentially the same, something that doesn't happen naturally, and they will possibly confuse various algorithms.

Generate the assembly with just the CLR reads, and use the CCS (not subreads) for polishing; the CCS reads are probably of higher quality than the Canu corrected reads.

After that, you could try a second assembly with the corrected CLR and the CCS, but as Sergey pointed out, with ~40x long reads, the CCS probably won't help contiguity much.

Basically, we know what ~40x of CLR reads assembles like, so do that one first.  We don't know what the other mixes will do, and suspect CLR+CCS won't improve much, and CLR+CCSsubread won't work.

**Comment by lamz138138** [comment]
OK, thanks for your reply and suggestion very much!

Best wishes!
